tests/subresource_api_test.go

```diff
@@ -73,6 +74,50 @@ var _ = Describe("Subresource Api", func() {
 			}, 15)
 		})
 	})
+
+	Describe("VirtualMachine subresource", func() {
+		Context("with a restart endpoint", func() {
+			It("should restart a VM", func() {
+				vm := NewRandomVirtualMachine(tests.NewRandomVMI(), false)
+				vm, err := virtCli.VirtualMachine(tests.NamespaceTestDefault).Create(vm)
+				Expect(err).NotTo(HaveOccurred())
+
+				tests.StartVirtualMachine(vm)
+				vmi, err := virtCli.VirtualMachineInstance(tests.NamespaceTestDefault).Get(vm.Name, &metav1.GetOptions{})
+				Expect(err).NotTo(HaveOccurred())
+				Expect(vmi.Status.Phase).To(Equal(v1.Running))
+
+				err = virtCli.VirtualMachine(tests.NamespaceTestDefault).Restart(vm.Name)
+				Expect(err).NotTo(HaveOccurred())
+
+				Eventually(func() v1.VirtualMachineInstancePhase {
+					newVMI, err := virtCli.VirtualMachineInstance(tests.NamespaceTestDefault).Get(vm.Name, &metav1.GetOptions{})
+					Expect(err).ToNot(HaveOccurred())
```

Comment:
I'm wondering if an error might be ok at this place, when the old VMI is deleted, but the new one not created yet?

Reply:
that's a good point, it's possible that error will occur here and that's ok. I will fix it.

Reply:
done